src/lib/y2users/equatable.rb

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Equatable
+  def self.included(base)
+    base.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    attr_reader :eql_attrs
+
+    def eql_attr(*names)
+      @eql_attrs ||= []
+      @eql_attrs += names
+    end
+  end
+
+  # Other strategies can be used in order to generate the hash key, like creating a Hash object and
```

Comment:
Originally I was thinking on creating a Hash with the attributes / values calling hash over it but we could also call hash over a sequence

https://ruby-doc.org/core-2.6.3/Enumerator/ArithmeticSequence.html#method-i-hash

Similar or basically what the Equatable gem does.

https://github.com/piotrmurach/equatable/blob/master/lib/equatable.rb#L102

Did you get inspired by that gem already?.

Reply:
I did not know it :)

Reply:
Me neither ;)